ranch/storage.go

```diff
@@ -482,21 +544,21 @@ func (s *Storage) persistDynamicResourceLifeCycles(dRLCToUpdate, dRLCToAdd, dRLC
 			// Mark this DRLC as pending deletion by setting min and max count to zero.
 			dRLC.Spec.MinCount = 0
 			dRLC.Spec.MaxCount = 0
-			dRLCToUpdate = append(dRLCToUpdate, dRLC)
-		}
-	}
-
-	for _, DRLC := range dRLCToAdd {
-		logrus.Infof("Adding resource type life cycle %s", DRLC.Name)
-		if err := s.AddDynamicResourceLifeCycle(&DRLC); err != nil {
-			errs = append(errs, err)
-		}
-	}
-
-	for _, dRLC := range dRLCToUpdate {
-		logrus.Infof("Updating resource type life cycle %s", dRLC.Name)
-		if _, err := s.UpdateDynamicResourceLifeCycle(&dRLC); err != nil {
-			errs = append(errs, err)
+			if err := s.client.Update(s.ctx, &dRLCToDelelete[idx]); err != nil {
+				errs = append(errs, err)
+				continue
+			}
+			// Make sure we have this change in our cache
```

Comment:
should we just put this logic in `UpdateDynamicResourceLifeCycle`, and then reduce some of the code-duplication? (I guess figuring out the right condition to check is tricky...)

couldn't we run into similar issues where an update to a DRLC (e.g. changing the mincount/maxcount to something nonzero) results in us creating and then immediately deleting a dynamic resource?

I guess it might increase the latency for syncs, but hopefully not by much?

Reply:
Done (We have this logic twice, once in syncDynamicResourceLifeCycles and once in UpdateAllDynamicResources. We should probably unexport the latter and remove those bits there, as they are redundant now)

Reply:
can you replace the duplicated logic in this function now with a call to `UpdateDynamicResourceLifeCycle`? i.e. replace the `s.client.Update` call (after setting MinCount and MaxCount to 0) with `UpdateDynamicResourceLifeCycle(&dRLCToDelelete[idx])`?

(also I just noticed that typo on `Delete` in `dRLCToDelelete`)

Reply:
Done and fixed the typo